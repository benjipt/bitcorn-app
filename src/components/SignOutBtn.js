import React from 'react'

export default function SignOutBtn({ handleLogout }) {
    return (
        <div>
            <button onClick={ handleLogout } className="btn btn-outline-dark">Sign Out</button>
        </div>
    )
}
