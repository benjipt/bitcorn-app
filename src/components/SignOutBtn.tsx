interface SignOutBtnProps {
  handleLogout: () => void;
}

export default function SignOutBtn({ handleLogout }: SignOutBtnProps) {
  return (
    <div>
      <button
        onClick={handleLogout}
        className='btn btn-outline-dark'
        data-testid='SignOutBtn-1'>
        Sign Out
      </button>
    </div>
  );
}
