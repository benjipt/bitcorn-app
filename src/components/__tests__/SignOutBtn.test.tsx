import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SignOutBtn from '../SignOutBtn';

describe('SignOutBtn', () => {
  it('should render SignOutButton component', () => {
    // Renders SignOutBtn Component
    render(<SignOutBtn handleLogout={() => {}} />);
    // Grabs component from render
    const SignOutBtnComponent = screen.getByTestId('SignOutBtn-1');
    // Tests for expected component
    expect(SignOutBtnComponent).toBeInTheDocument();
  });

  it('calls handleLogout prop when called', () => {
    // Mocks function passed as prop
    const handleLogout = vi.fn();
    // Renders component with handleLogout prop
    render(<SignOutBtn handleLogout={handleLogout} />);
    // Grabs component from render
    const SignOutBtnComponent = screen.getByTestId('SignOutBtn-1');
    // Simulates click event
    fireEvent.click(SignOutBtnComponent);
    // Tests that mock function was called once
    expect(handleLogout).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    // Render component tree
    const tree = renderer
      .create(<SignOutBtn handleLogout={() => {}} />)
      .toJSON();
    // Tests for matching snapshot
    expect(tree).toMatchSnapshot();
  });
});
