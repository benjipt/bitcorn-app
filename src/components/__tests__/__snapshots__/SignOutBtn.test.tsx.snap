// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignOutBtn > matches snapshot 1`] = `
<div>
  <button
    className="btn btn-outline-dark"
    data-testid="SignOutBtn-1"
    onClick={[Function]}
  >
    Sign Out
  </button>
</div>
`;

exports[`SignOutBtn matches snapshot 1`] = `
<div>
  <button
    className="btn btn-outline-dark"
    data-testid="SignOutBtn-1"
    onClick={[Function]}
  >
    Sign Out
  </button>
</div>
`;
